src/lib/loaders/api/loader_without_cache_factory.ts

```diff
@@ -11,32 +13,40 @@ import { DataLoaderKey } from "./index"
  * @param {string} _apiName the name of the API service
  */
 
-export const uncachedLoaderFactory = (
-  api: (route: string, params) => Promise<any>,
-  apiName: string
+export const uncachedLoaderFactory = <T = any>(
+  api: API,
+  _apiName: string,
+  apiOptions: APIOptions = {}
 ) => {
-  const apiLoaderFactory = (path, options: any | null) => {
-    // If you use gravity as the api here, then options will get interpreted as
-    // an accessToken, so you have to explicitly pass null
-    const loader = new DataLoader<DataLoaderKey, any>(
+  const apiLoaderFactory = (path, params = {}) => {
+    const loader = new DataLoader<DataLoaderKey, T | { body: T; headers: any }>(
       (keys) =>
-        Promise.all(
-          keys.map(({ key, apiOptions }) => {
-            if (apiOptions) {
-              throw new Error("A uncachedLoader does not accept API options.")
-            }
-
+        Promise.all<any>(
+          keys.map(({ key }) => {
             const reduceData = ({ body, headers }) =>
-              options && options.headers ? { body, headers } : body
+              apiOptions && apiOptions.headers ? { body, headers } : body
+
+            const callApi = () =>
+              api(key, null, apiOptions)
+                .then(reduceData)
+                .catch((error) => {
```

Comment:
Will remove this catch. Better to just blow up here instead of throwing.

Reply:
Actually we need to finish implementing this to log exact errors into GraphiQL! This was only done on the auth loader factory; just needed to finish up here: https://github.com/artsy/metaphysics/blob/main/src/lib/loaders/api/loader_with_authentication_factory.ts#L67-L78

Reply:
Cool! Thanks for the example. 👍🏽 Adding now.

Reply:
Done. Much better error messaging this way. 😅

Reply:
ty!